src/urllib3/response.py

```diff
@@ -811,7 +811,9 @@ def _fp_read(
                 return self._fp.read1(size)
         elif (
             (size > c_int_max)
-            or (self.length_remaining and self.length_remaining > c_int_max)
+            or (
+                size < 0 and self.length_remaining and self.length_remaining > c_int_max
+            )
```

Comment:
Interested to know opinions about this deliberate change in semantics.

I feel as though the code should change to be doing the same chunked reading workaround when `length_remaining is None`.